### PR TITLE
Sprint 9 solution http api

### PR DESCRIPTION
**Pull request by lasfin**

### CODE REVIEW

**Review by avfyodorov — COMMENTED** [review]
Добрый вечер, Евгений! 

Хорошая работа, единственное, два теста не проходят - 

[ERROR] Failures: 
[ERROR]   FileBackedTaskServiceTest.shouldLoadTasksFromExternalFile:111 expected: <2> but was: <1>
[ERROR] Errors: 
[ERROR]   FileBackedTaskServiceTest.shouldLoadExternalFileByStaticMethod:147 » DateTimeParse Text 'none
2' could not be parsed at index 0

**Comment by lasfin** [comment]
> Добрый вечер, Евгений!
> 
> Хорошая работа, единственное, два теста не проходят -
> 
> [ERROR] Failures: [ERROR] FileBackedTaskServiceTest.shouldLoadTasksFromExternalFile:111 expected: <2> but was: <1> [ERROR] Errors: [ERROR] FileBackedTaskServiceTest.shouldLoadExternalFileByStaticMethod:147 » DateTimeParse Text 'none 2' could not be parsed at index 0

Добрый вечер!
А где не проходят?
Вот же пайплайн, все прошло: https://github.com/lasfin/java-kanban/actions/runs/12018338166/job/335

Снимок экрана 2024-11-26 в 17 54 21
02604004?pr=9

File: src/main/java/services/task/FileBackedTaskService.java [hunk 1]
```diff
@@ -174,21 +174,27 @@ public Task fromString(String str) {
...
 
     @Override
-    public void addSubtask(Subtask subtask) {
-        super.addSubtask(subtask);
+    public Subtask addSubtask(Subtask subtask) {
+        Subtask st = super.addSubtask(subtask);
         save();
+
+        return st;
     }
```
Comment (avfyodorov) [inline src/main/java/services/task/FileBackedTaskService.java:199]: К сожалению, тестовый класс  FileBackedTaskServiceTest в pull request не попал, поэтому пишу здесь. 

    @Test
    public void shouldLoadExternalFileByStaticMethod() {
        String fileName = "temp.csv";
……………..

java.time.format.DateTimeParseException: Text 'none
2' could not be parsed at index 0

	at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2108)
	at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:2010)
	at java.base/java.time.LocalDateTime.parse(LocalDateTime.java:494)
	at services.task.FileBackedTaskService.fromString(FileBackedTaskService.java:151)
	at services.task.FileBackedTaskService.loadFromFile(FileBackedTaskService.java:55)
	at services.task.FileBackedTaskService.loadOrNotify(FileBackedTaskService.java:42)
	at services.task.FileBackedTaskService.<init>(FileBackedTaskService.java:32)
	at services.task.FileBackedTaskService.loadFromFile(FileBackedTaskService.java:36)
	at services.task.FileBackedTaskServiceTest.shouldLoadExternalFileByStaticMethod(FileBackedTaskServiceTest.java:147)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)

File: src/main/java/services/task/FileBackedTaskService.java [hunk 2]
```diff
@@ -174,21 +174,27 @@ public Task fromString(String str) {
...
 
     @Override
```
Comment (avfyodorov) [inline src/main/java/services/task/FileBackedTaskService.java:200]: и второй

    @Test
    public void shouldLoadTasksFromExternalFile() {
        String fileName = "temp.csv";
        try {
..............

org.opentest4j.AssertionFailedError: 
Expected :2
Actual   :1
<Click to see difference>


	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:531)
	at services.task.FileBackedTaskServiceTest.shouldLoadTasksFromExternalFile(FileBackedTaskServiceTest.java:111)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)

**Review by avfyodorov — COMMENTED** [review]
Добрый вечер! 

Не проходят у меня на компьютере. Всё-таки, тесты не должны валиться.

**Review by avfyodorov — COMMENTED** [review]
Добрый вечер, Евгений!

Всё в порядке, замечаний нет. 
Работа принята.

**Pull request merged**